### PR TITLE
PLACES-257 Add consented_to_publishing_at field on cases table

### PR DESCRIPTION
**Pull request by MarshallMcCoy**
Jira Ticket: https://pathcheck.atlassian.net/browse/PLACES-257

- Add a new field on the `cases` database table to track when a user consented to having their data published
- Update the existing `consentToPublish` controller action in the cases controller to set the value on the new field in the database when called

### CODE REVIEW

File: db/models/cases.js [hunk 1]
```diff
@@ -37,31 +37,45 @@ class Service extends BaseService {
...
 
+  /**
+   * Consent To Publish
+   *
+   * @method consentToPublishing
+   * @param {Number} case_id
+   * @return {Object}
+   */
+  async consentToPublishing(case_id) {
+    const result = await this.updateOne(case_id, { consented_to_publishing_at: new Date() });
```
Comment (jvisenti) [inline db/models/cases.js:48]: Would prefer `this.database.fn.now()` over `new Date()`

File: db/models/cases.js [hunk 2]
```diff
@@ -37,31 +37,45 @@ class Service extends BaseService {
...
+   * @method moveToStaging
    * @param {Number} case_id
-   * @return {Array}
+   * @return {Object}
    */
   async moveToStaging(case_id) {
-    const results = await this.updateOne(case_id, { state: 'staging' });
-    if (results) {
-      return this._mapCase(results);
+    const result = await this.updateOne(case_id, { state: 'staging' });
```
Comment (jvisenti) [inline db/models/cases.js:62]: This is currently in conflict with dev, don't forget to keep the `staged_at` change from dev as you resolve the conflict.

File: test/integration/case.test.js [hunk 3]
```diff
@@ -144,19 +144,41 @@ describe('Case', () => {
...
+
+        result.error.should.be.false;
+        result.should.have.status(200);
+        result.body.should.be.a('object');
+        result.body.should.have.property('case');
+        result.body.case.should.be.a('object');
+        result.body.case.should.have.property('caseId');
+        result.body.case.should.have.property('state');
+        result.body.case.should.have.property('updatedAt');
+        result.body.case.should.have.property('expiresAt');
```
Comment (jvisenti) [inline test/integration/case.test.js:179]: Add a check that `consentedToPublishingAt` exists on the returned case object

File: db/models/cases.js (hunk 1 above)
Comment (jvisenti) [inline db/models/cases.js:48]: Also need to add `consented_to_publishing_at -> consentedToPublishingAt` mapping in `_mapCase`. If it's not supposed to be exposed in the returned object, should still `delete itm.consented_to_publishing_at;`.

File: test/integration/case.test.js (hunk 3 above)
Comment (MarshallMcCoy) [reply test/integration/case.test.js:179]: The frontend has no use for the this piece of data at the moment. Its just something we store on the backend for record keeping. For that reason I feel that we should leave it out of the response payloads.

Comment (jvisenti) [reply test/integration/case.test.js:179]: Should `stagedAt` have similar treatment? Or is this use by the frontend?

Comment (MarshallMcCoy) [reply test/integration/case.test.js:244]: The frontend will be using the `stagedAt` field.

Comment (jvisenti) [reply test/integration/case.test.js:244]: 👍

**Pull request merged**